### PR TITLE
Temporary test indices

### PR DESCRIPTION
**Pull request by paul-butcher**
At this point, temporary test indices are created using the New Method, but there has been no change to the actual behaviour of the application.

The test suite (and application) both still use Elastic4s to index documents and execute queries.

### CODE REVIEW

**Comment by paul-butcher** [comment]
https://github.com/wellcomecollection/catalogue-pipeline/issues/2369

File: common/internal_model/src/test/scala/weco/catalogue/internal_model/fixtures/index/IndexFixturesBase.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  protected def getConfig(mappings: String, analysis: String) =
+    s"""{"mappings":${Source
+        .fromResource(mappings)(Codec.UTF8)
+        .mkString},
+      "settings":{"analysis":   
+      ${Source
+        .fromResource(analysis)(Codec.UTF8)
+        .mkString}
+      }}
+      """
```
Comment (alexwlchan) [inline common/internal_model/src/test/scala/weco/catalogue/internal_model/fixtures/index/IndexFixturesBase.scala:18]: Let's extract the two calls to `Source.fromResource` as variables, so it's easier to see what the final JSON document looks like. Similar for the method on L20–27.

```suggestion
  protected def getConfig(mappings: String, analysis: String) = {
    val mappings = Source
      .fromResource(mappings)(Codec.UTF8)
      .mkString
      
    val analysis = Source
      .fromResource(analysis)(Codec.UTF8)
      .mkString
  
    s"""{"mappings": $mappings, "settings": {"analysis": $analysis}}"""
  }
```

File: common/internal_model/src/test/scala/weco/catalogue/internal_model/fixtures/index/ImagesIndexFixtures.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    withLocalUnanalysedJsonStore[R]
+
+  def withLocalImagesIndex[R]: Fixture[Index, R] = {
+    withLocalElasticSearchIndex[R](config =
+      getConfig(
+        mappings = "mappings.images_indexed.2023-05-26.json",
+        analysis = "analysis.works_indexed.2023-05-26.json"
+      )
+    )
+  }
```
Comment (alexwlchan) [inline common/internal_model/src/test/scala/weco/catalogue/internal_model/fixtures/index/ImagesIndexFixtures.scala:19]: this is really nice 🌵

File: common/internal_model/src/test/scala/weco/catalogue/internal_model/fixtures/index/IndexFixturesE4S.scala [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import org.scalatest.Assertion
+import weco.json.JsonUtil.toJson
+import weco.catalogue.internal_model.image.{Image, ImageState}
+import weco.catalogue.internal_model.work.{Work, WorkState}
+import org.scalatest.matchers.should.Matchers
+
+/*
+ * Temporary shimmy kind of thing that still uses Elastic4s to do some elasticy things
+ *
+ * These are just copied out of the existing code and are waiting to be rewritten.
```
Comment (alexwlchan) [inline common/internal_model/src/test/scala/weco/catalogue/internal_model/fixtures/index/IndexFixturesE4S.scala:22]: Not trying to rewrite the entire world in one go 👍

File: common/internal_model/src/test/scala/weco/catalogue/internal_model/index/WorksIndexConfigTest.scala [hunk 4]
```diff
@@ -43,7 +44,7 @@ class WorksIndexConfigTest
 
   describe("indexing different works with every type of WorkState") {
     it("WorkState.Source") {
-      withLocalElasticsearchIndex(config = WorksIndexConfig.source) {
+      withLocalSourceWorksIndex {
```
Comment (alexwlchan) [inline common/internal_model/src/test/scala/weco/catalogue/internal_model/index/WorksIndexConfigTest.scala:47]: 🤌

File: common/internal_model/src/test/scala/weco/catalogue/internal_model/matchers/EventuallyInElasticsearch.scala [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import org.scalatest.Assertion
+import weco.elasticsearch.model.IndexId
+import weco.json.JsonUtil.toJson
+import org.scalatest.matchers.should.Matchers
+import weco.catalogue.internal_model.image.{Image, ImageState}
+import weco.catalogue.internal_model.work.{Work, WorkState}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+//TODO: Consider turning this into a matcher?
```
Comment (alexwlchan) [inline common/internal_model/src/test/scala/weco/catalogue/internal_model/matchers/EventuallyInElasticsearch.scala:18]: Would definitely be nice, we haven't leaned much on matchers but they're super handy.

File: common/pipeline_storage/src/test/scala/weco/pipeline_storage/ImageIndexableTest.scala [hunk 6]
```diff
@@ -107,7 +106,7 @@ class ImageIndexableTest
   private def withImagesIndexAndIndexer[R](
     testWith: TestWith[(Index, ElasticIndexer[Image[ImageState.Augmented]]), R]
   ) =
-    withLocalAugmentedImageIndex { index =>
+    withLocalAugmentedImageIndex { index:Index =>
```
Comment (alexwlchan) [inline common/pipeline_storage/src/test/scala/weco/pipeline_storage/ImageIndexableTest.scala:109]:
```suggestion
    withLocalAugmentedImageIndex { index: Index =>
```

why hasn't the autoformatter picked this up?

(there were a few other spots)

**Review by alexwlchan — APPROVED** [review]
This is a really nice piece of work. The new fixtures are easy to follow, and having named helpers for e.g. `withLocalIdentifiedWorksIndex` makes those tests a bit more readable.

I've put a couple of suggestions inline, but they're all pretty small – happy for you to merge as soon as those are addressed; I don't need to look at this again.

**Pull request merged**